tests/test_estimators/test_quantile_regression.py

```diff
@@ -0,0 +1,111 @@
+"""Test the QuantileRegression."""
+
+import numpy as np
+import pytest
+
+from sklego.linear_model import QuantileRegression
+from sklego.testing import check_shape_remains_same_regressor
+from sklego.common import flatten
+from tests.conftest import general_checks, nonmeta_checks, select_tests, regressor_checks
+
+test_batch = [
+    (np.array([0, 0, 3, 0, 6]), 3),
+    (np.array([1, 0, -2, 0, 4, 0, -5, 0, 6]), 2),
+    (np.array([4, -4]), 0),
+]
+
+
+def _create_dataset(coefs, intercept, noise=0.0):
+    np.random.seed(0)
+    X = np.random.randn(10000, coefs.shape[0])
+    y = X @ coefs + intercept + noise * np.random.randn(10000)
+
+    return X, y
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_coefs_and_intercept__no_noise(coefs, intercept):
+    """Regression problems without noise."""
+    X, y = _create_dataset(coefs, intercept)
+    quant = QuantileRegression()
+    quant.fit(X, y)
+
+    assert quant.score(X, y) > 0.99
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+def test_score(coefs, intercept):
+    """Tests with noise on an easy problem. A good score should be possible."""
+    X, y = _create_dataset(coefs, intercept, noise=1.0)
+    quant = QuantileRegression()
+    quant.fit(X, y)
+
+    assert quant.score(X, y) > 0.9
+
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+@pytest.mark.parametrize("quantile", np.linspace(0, 1, 7))
+def test_quantile(coefs, intercept, quantile):
+    """Tests with noise on an easy problem. A good score should be possible."""
+    X, y = _create_dataset(coefs, intercept, noise=1.0)
+    quant = QuantileRegression(quantile=quantile)
+    quant.fit(X, y)
+
+    np.testing.assert_almost_equal((quant.predict(X)>=y).mean(), quantile, decimal=2)
```

Comment:
I didn't have the time to download and go fully into the numerics myself, but from my end ... the fact that this test passes gives me enough confidence that it's green to merge. 

Before I do that. @MBrouns just to check if you had any comments?